### PR TITLE
feat(condo): DOMA-7577 invoice form

### PR DESCRIPTION
**Pull request by nomerdvadcatpyat**
Added create and update invoice forms

https://github.com/open-condo-software/condo/assets/52532264/9a094748-a099-4d66-a618-4d23c1c9296b

### CODE REVIEW

File: apps/condo/domains/marketplace/components/Invoice/UpdateInvoiceForm.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        where: {
+            organization: { id: organization.id },
+        },
+    })
+
+    const [registerMultiPayment] = useMutation(REGISTER_MULTI_PAYMENT_MUTATION)
+
+    const updateInvoiceAction = Invoice.useUpdate({}, async () => {
+        // replace after create marketplace/index page
+        // await router.push('/ticket')
```
Comment (AleX83Xpert) [inline apps/condo/domains/marketplace/components/Invoice/UpdateInvoiceForm.tsx:40]: Already created
```suggestion
        await router.push('/marketplace?tab=bills')
```

File: apps/condo/domains/resident/schema/GetResidentExistenceByPhoneAndAddressService.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+const access = require('@condo/domains/resident/access/GetResidentExistenceByPhoneAndAddressService')
+const { RESIDENT } = require('@condo/domains/user/constants/common')
+
+
+const GetResidentExistenceByPhoneAndAddressService = new GQLCustomSchema('GetResidentExistenceByPhoneAndAddressService', {
+    types: [
+        {
+            access: true,
+            type: 'input GetResidentExistenceByPhoneAndAddressInput { dv: Int!, sender: JSON!, phone: String!, propertyId: ID!, unitName: String!, unitType: BuildingUnitSubType! }',
```
Comment (AleX83Xpert) [inline apps/condo/domains/resident/schema/GetResidentExistenceByPhoneAndAddressService.js:14]:
```suggestion
            type: 'input GetResidentExistenceByPhoneAndAddressInput { dv: Int!, sender: SenderFieldInput!, phone: String!, propertyId: ID!, unitName: String!, unitType: BuildingUnitSubType! }',
```

**Review by AleX83Xpert — APPROVED** [review]

File: apps/condo/domains/marketplace/components/Invoice/UpdateInvoiceForm.tsx (hunk 1 above)
Comment (nomerdvadcatpyat) [reply apps/condo/domains/marketplace/components/Invoice/UpdateInvoiceForm.tsx:40]: submit update form should return to invoice card, in create form fix it

File: apps/condo/domains/marketplace/components/Invoice/BaseInvoiceForm.tsx [hunk 3]
```diff
@@ -0,0 +1,1095 @@
...
+        marketItemGroups.sort((a, b) => a.key > b.key ? 1 : -1)
+
+        return marketItemGroups
+    }, [marketPriceScopes])
+
+    const flatMarketOptions = useMemo(() => marketItemGroups.flatMap(group => group.options), [marketItemGroups])
+
+    const moneyRender = useMemo(() => getMoneyRender(intl, FromMessage), [FromMessage, intl])
+
+    console.log('breakpoints', breakpoints)
```
Comment (VKislov) [inline apps/condo/domains/marketplace/components/Invoice/BaseInvoiceForm.tsx:443]: rm

File: apps/condo/domains/marketplace/components/Invoice/BaseInvoiceForm.tsx [hunk 4]
```diff
@@ -0,0 +1,1095 @@
...
+                                            name={[marketItemForm.name, 'count']}
+                                            required
+                                            labelAlign='left'
+                                            labelCol={{ span: 24 }}
+                                            rules={[requiredValidator]}
+                                            initialValue={1}
+                                        >
+                                            <Select
+                                                disabled={disabled}
+                                                options={[...Array(50).keys() ].map( i => ({
```
Comment (VKislov) [inline apps/condo/domains/marketplace/components/Invoice/BaseInvoiceForm.tsx:511]: It's not memoized. It seems that with each rerender this array and label, key will be created. Perhaps it’s worth removing this from the layout into the component itself and memoizing it.
ps: minor, does not affect approval

**Review by VKislov — APPROVED** [review]

**Pull request merged**